src/factories/routePathFactory.ts

```diff
@@ -1,18 +1,39 @@
 import { INode, IRoutePath } from '../models';
 import NodeFactory from './nodeFactory';
+import HashHelper from '../util/hashHelper';
 
 class RoutePathFactory {
     // suunta to IRoutePath
-    public static createRoutePath = (suunta: any, isVisible:boolean): IRoutePath => {
+    public static createRoutePath = (
+        routeId: string,
+        suunta: any,
+        isVisible:boolean,
+    ): IRoutePath => {
+        const madeUpId = HashHelper.getHashFromString(
+            [
+                routeId,
+                suunta.suunimi,
+                suunta.suuvoimast,
+                suunta.suuvoimviimpvm,
+                suunta.suusuunta,
+            ].join('-'),
+        ).toString();
+
         const nodes:INode[]
         = suunta.reitinlinkkisByReitunnusAndSuuvoimastAndSuusuunta.edges
-            .map((node: any) => NodeFactory.createNode(node.node));
+            .map((node: any) => NodeFactory.createNode(madeUpId, node.node));
+
+        const coordinates = JSON.parse(suunta.geojson).coordinates;
+        const positions = coordinates.map((coor: [number, number]) => [coor[1], coor[0]]);
 
         return <IRoutePath>{
+            routeId,
             nodes,
+            positions,
+            internalRoutePathId: madeUpId,
```

Comment:
Here it is as "internalRoutePathId" but in NodeFactory its "routePathId". We need consistency, use the same name everywhere.